backend/fms_core/tests/test_services/sample_next_step_test.py

```diff
@@ -32,22 +32,71 @@ def setUp(self):
 
     def test_queue_sample_to_study_workflow_default(self):
         sample_next_step, errors, warnings = queue_sample_to_study_workflow(self.sample, self.study)
+        is_queued, errors, warnings = is_sample_queued_in_study(self.sample, self.study)
         self.assertEqual(errors, [])
         self.assertEqual(warnings, [])
         self.assertTrue(isinstance(sample_next_step, SampleNextStep))
         self.assertEqual(sample_next_step.step_order.order, 1)
         self.assertEqual(sample_next_step.study, self.study)
 
+        #Test sample queued
+        self.assertTrue(is_queued)
+
     def test_queue_sample_to_valid_step(self):
         sample_next_step, errors, warnings = queue_sample_to_study_workflow(self.sample, self.study, 3)
-        self.assertEqual(errors, [])
-        self.assertEqual(warnings, [])
         self.assertTrue(isinstance(sample_next_step, SampleNextStep))
         self.assertEqual(sample_next_step.step_order.order, 3)
         self.assertEqual(sample_next_step.sample, self.sample)
 
+        #Test sample queued
+        is_queued, errors, warnings = is_sample_queued_in_study(self.sample, self.study, 3)
+        self.assertTrue(is_queued)
+        self.assertEqual(errors, [])
+        self.assertEqual(warnings, [])
+
     def test_queue_sample_to_invalid_step(self):
         sample_next_step, errors, warnings = queue_sample_to_study_workflow(self.sample, self.study, 8)
         self.assertEqual(errors, ["Order must be a positive integer between 1 and 7."])
         self.assertEqual(warnings, [])
         self.assertIsNone(sample_next_step)
+
+        #Test sample queued
+        is_queued, errors, warnings = is_sample_queued_in_study(self.sample, self.study)
+        self.assertFalse(is_queued)
+
+    def test_dequeue_sample_from_valid_step(self):
+        # Queue sample first and test it
+        sample_next_step, errors, warnings = queue_sample_to_study_workflow(self.sample, self.study)
+        is_queued, errors, warnings = is_sample_queued_in_study(self.sample, self.study)
```

Comment:
you should not test using another service as much as possible. This will just lead to cascade errors of the testing service as an error.